packages/node-cli/src/core/sdksetup/ProxyAgent.js

```diff
@@ -0,0 +1,92 @@
+/*
+ ** Copyright (c) 2021 Oracle and/or its affiliates.  All rights reserved.
+ ** Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+ */
+const http = require('http');
+const { URL } = require('url');
+const tls = require('tls');
+const net = require('net');
+
+class ProxyAgent {
+	constructor(proxyString, options) {
+		this.proxyString = proxyString;
+		this.options = options;
+		this.initProxyURL();
+	}
+
+	initProxyURL() {
+		// throws error if proxyString is invalid
+		const proxy = new URL(this.proxyString);
+
+		proxy.host = proxy.hostname || proxy.host;
+		proxy.port = +proxy.port || (proxy.protocol.toLowerCase() === 'https:' ? 443 : 80);
+		this.proxyURL = proxy;
+	}
+
+	addRequest(req, options) {
+		req.shouldKeepAlive = false;
+
+		this.createConnection(options)
+			.then((socket) => {
+				req.onSocket(socket);
+			})
+			.catch((err) => {
+				req.emit('error', err);
+			});
+	}
+
+	createConnection(options) {
```

Comment:
It's just a matter of taste, but I would rename options into parameters/params so it's not confused with the class property

Reply:
I like the suggestion :-) . I'm renaming the class options to agentOptions , which will accomplish the same goal.